extensions/platform-http/runtime/src/main/java/org/apache/camel/quarkus/component/platform/http/runtime/QuarkusPlatformHttpConsumer.java

```diff
@@ -111,13 +111,18 @@ protected void doStart() throws Exception {
 
         newRoute.handler(
             ctx -> {
+                final Exchange exchange = toExchange(ctx);
                 try {
-                    final Exchange e = toExchange(ctx);
-                    getProcessor().process(e);
-                    writeResponse(ctx, e, getEndpoint().getHeaderFilterStrategy());
+                    createUoW(exchange);
+                    getAsyncProcessor().process(
+                        exchange,
+                        doneSync -> writeResponse(ctx, exchange, getEndpoint().getHeaderFilterStrategy())
+                    );
                 } catch (Exception e) {
                     LOG.debugf(e, "Could not handle '%s'", path);
                     ctx.fail(e);
```

Comment:
You may need to do something like: `getExceptionHandler().handleException(e)` like https://github.com/apache/camel/blob/master/components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowConsumer.java#L171

Reply:
Yes, I'd see `getExceptionHandler().handleException(e)` as a good replacement for the current `LOG.debugf(e, "Could not handle '%s'", path)` because the default `LoggingExceptionHandler` just does some logging and the message it logs is actually better than `Could not handle '%s'`.

But we still need to mark the Vert.x `RoutingContext` as failed somehow, because otherwise it will hang forever, won't it? We have no custom ExceptionHandler on the Camel side that would do this and I see no good reason to create one, because calling `ctx.fail(e)` is much less code. WDYT?

Reply:
yeah ctx.fail is fine